client/lib/i18n-utils/utils.js

```diff
@@ -19,12 +19,25 @@ function getPathParts( path ) {
 }
 
 const i18nUtils = {
+	/**
+	 * Checks if provided locale is a default one.
+	 *
+	 * @param {string} locale - locale slug (eg: 'fr')
+	 * @return {boolean} true when the default locale is provided
+	 */
+	isDefaultLocale: function( locale ) {
+		return locale === config( 'i18n_default_locale_slug' );
+	},
+
 	getLanguage: function( langSlug ) {
 		let language;
+
 		if ( localeRegex.test( langSlug ) || localeWithRegionRegex.test( langSlug ) ) {
-			language = find( config( 'languages' ), { langSlug: langSlug } ) ||
-				find( config( 'languages' ), { langSlug: langSlug.substring( 0, 2 ) } );
+			language =
+				find( config( 'languages' ), { langSlug } ) ||
+				find( config( 'languages' ), { langSlug: langSlug.split( '-' )[ 0 ] } );
```

Comment:
Check [this config](https://github.com/Automattic/wp-calypso/blob/e1e5f75039a838fdaf8f9e875d4f1e1f0f97d399/config/_shared.json#L39). We have two locales like:
```js
{ "value": 3, "langSlug": "ar", "name": "العربية", "wpLocale": "ar", "rtl": true, "popular": 16 },
{ "value": 419, "langSlug": "arc", "name": "ܕܥܒܪܸܝܛ", "wpLocale": "" },
```
Now image you pass `arc-xxx` as locale. With substring it would return `ar` and it would display a completely different language than you would expect.

There was a very strange behavior with `log-in`, it matches locale pattern and with substring it was reduced to `lo`, which is valid locale :P `log` locale doesn't exist, so we are good to go. Still it may fail if we ever provide top level url which is equal to locale, but it is less likely to happen :)